hyperspace/cosmos/src/lib.rs

```diff
@@ -153,24 +181,85 @@ where
 	/// Construct a tendermint client state to be submitted to the counterparty chain
 	pub async fn construct_tendermint_client_state(
 		&self,
-	) -> Result<(AnyClientState, AnyConsensusState), Error>
+	) -> Result<(AnyClientState, AnyConsensusState), Error> {
+		let latest_height_timestamp = self.latest_height_and_timestamp().await.unwrap();
+		let client_state = TmClientState::<HostFunctionsManager>::new(
+			self.chain_id.clone(),
+			TrustThreshold::default(),
+			Duration::new(64000, 0),  // Set to a default value
+			Duration::new(128000, 0), // Set to a default value
+			Duration::new(5, 0),      // Set to a default value
+			latest_height_timestamp.0,
+			ProofSpecs::default(),
+			vec!["upgrade".to_string(), "upgradedIBCState".to_string()],
+		)
+		.map_err(|e| Error::from(format!("Invalid client state {}", e)))?;
+
+		let target_height = TmHeight::try_from(latest_height_timestamp.0.revision_height)
+			.map_err(|e| Error::from(e.to_string()))?;
+		let trusted_block = self
+			.light_provider
+			.fetch_light_block(AtHeight::At(target_height))
+			.await
+			.map_err(|e| {
+				Error::from(format!("Failed to fetch light block from light provider: {:?}", e))
+			})?;
+
+		let consensus_state = TmConsensusState::from(trusted_block.signed_header.header);
+
+		Ok((
+			AnyClientState::Tendermint(client_state),
+			AnyConsensusState::Tendermint(consensus_state),
+		))
+	}
+
+	/// Construct a tendermint light client to communicate with the full node
+	pub async fn _construct_tendemint_light_client<HostFunctions: CryptoProvider>(
```

Comment:
We can always import the `ics07-tendermint` crate if we need to do any verification, so I doubt this would ever be needed here.

Reply:
Alright, it's removed https://github.com/ComposableFi/centauri/pull/139/commits/4cd48ad8a0b8a2d11504f24e6ac6806aca272da8